File: src/java_caesar/Decryption.java
import java.util.Scanner;

public class Decryption {
    private Scanner sc = new Scanner(System.in);
    private String message;
    private String decryptedMessage = "";
    private int key;

    public String getDecryptedMessage() {
        return decryptedMessage;
    }

    public Decryption(String message, int key) {
        this.message = message;
        this.key = key;
    }

    public Decryption() {
        this.key = 3;
    }

    public void cipherMessage() {
        System.out.print("Enter a message: ");
        message = sc.nextLine();

        System.out.print("Enter key: ");
        key = sc.nextInt();
    }

    public void decrypt() {
        decryptedMessage = getDecrypt(decryptedMessage, key, message);

        System.out.println("Decrypted message = " + decryptedMessage);
    }

    static String getDecrypt(String tempMessage, int i, String message) {
        for (int j = 0; j < message.length(); ++j) {
            char ch = message.charAt(j);

            if (ch >= 'a' && ch <= 'z') {
                ch = (char) (ch - i);

                if (ch < 'a') {
                    ch = (char) (ch + 'z' - 'a' + 1);
                }

                tempMessage += ch;
            } else if (ch >= 'A' && ch <= 'Z') {
                ch = (char) (ch - i);

                if (ch < 'A') {
                    ch = (char) (ch + 'Z' - 'A' + 1);
                }

                tempMessage += ch;
            } else {
                tempMessage += ch;
            }
        }
        return tempMessage;
    }
}

File: src/kotlin_caesar/Decryption.kt
package kotlin_caesar

import java.util.*

class Decryption {
    private val sc: Scanner = Scanner(System.`in`)
    private var message: String? = null
    var decryptedMessage = ""
        private set
    private var key: Int

    constructor(message: String?, key: Int) {
        this.message = message
        this.key = key
    }

    constructor() {
        key = 3
    }

    fun cipherMessage() {
        print("Enter a message: ")
        message = sc.nextLine()
        print("Enter key: ")
        key = sc.nextInt()
    }

    fun decrypt() {
        decryptedMessage = getDecrypt(decryptedMessage, key, message)
        println("Decrypted message = $decryptedMessage")
    }

    companion object {
        fun getDecrypt(tempMessage: String, i: Int, message: String?): String {
            var tempMessage = tempMessage
            for (element in message!!) {
                var ch = element
                when (ch) {
                    in 'a'..'z' -> {
                        ch = (ch.toInt() - i).toChar()
                        if (ch < 'a') {
                            ch = (ch + 'z'.toInt() - 'a'.toInt() + 1)
                        }
                        tempMessage += ch
                    }
                    in 'A'..'Z' -> {
                        ch = (ch.toInt() - i).toChar()
                        if (ch < 'A') {
                            ch = (ch + 'Z'.toInt() - 'A'.toInt() + 1)
                        }
                        tempMessage += ch
                    }
                    else -> {
                        tempMessage += ch
                    }
                }
            }
            return tempMessage
        }
    }
}